### PR TITLE
Add universe_domain field as a provider attribute

### PR DESCRIPTION
**Pull request by hao-nan-li**
Add universe_domain field as a provider attribute

Keep testing edge cases due to changed based URL

If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [ ] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:note
provider: added `universe_domain` attribute as a provider attribute
```

### CODE REVIEW

**Comment by hao-nan-li** [comment]
Note: This is WIP, I'll add description to it. Please feel free to give comments.

File: mmv1/third_party/terraform/tests/universe_template_test.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-provider-google/google/acctest"
+)
+func TestAccUniverseDomain(t *testing.T) {
+	t.Parallel()
```
Comment (rileykarson) [inline mmv1/third_party/terraform/tests/universe_template_test.go:11]: Note that we should skip or fail the test if key values are unset, with clear errors messages. That will mean adding a handler function for `GOOGLE_UNIVERSE_DOMAIN` (or a similar env var) and a handler (either a function or inline) for a custom domain until those are no longer required.

Comment (hao-nan-li) [reply mmv1/third_party/terraform/tests/universe_template_test.go:12]: Sure, I will leave this test skipped.

**Comment by hao-nan-li** [comment]
Still validating different integration tests failing scenarios. 

I'm a bit confused why all VCR tests passed, locally I'm getting a lot of failures due to corner cases.

**Comment by hao-nan-li** [comment]
Moved to https://github.com/GoogleCloudPlatform/magic-modules/pull/8657 due to tgc tests failures

**Pull request closed without merging**